core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -1255,6 +1277,9 @@ private void writeValue(Object value) throws IOException {
             generator.writeEndObject();
         } else if (value instanceof Map) {
             writeMap((Map) value);
+        } else if (value instanceof Path) {
+            //Path implements Iterable<Path> and causes endless recursion and a StackOverFlow if treated as an Iterable here
+            generator.writeString(value.toString());            
```

Comment:
Yes I was thinking the same thing, e.g. if it is relative, making it absolute during serialization would change the semantics, which is not something we want to do in serialization code. Still, having these issues is why I wasn't sure if we wanted to have dedicated support for Path, or have the caller decide what aspect of the Path should be serialized, and call XContentBuilder with the string it created. Some could could keep the relative path, others could make it absolute first, maybe also normalize, resolve symlinks, etc.

Overall, toString() felt as a good default, caller can still do the custom serialization if needed.